conanfile.py

```diff
@@ -91,6 +92,14 @@ def set_version(self):
         version = re.search(r'(\d+\.\d+\.\d+)', content).group(1)
         self.version = version.strip()
 
+    def checkMoldIsSupported(self):
+        """ Mold is only tested on Linux with gcc and clang. In future support for icc may be added. """
+        return self.settings.os == "Linux" and (self.settings.compiler == "clang" or self.settings.compiler == "gcc")
+    
+    def configure(self):
+        if not self.checkMoldIsSupported():
+            self.options.rm_safe("link_with_mold")
```

Comment:
So now if `mold` is not supported then we delete the option on any unsupported platforms (i.e. Windows, Mac and untested compilers on Linux) so that it cannot be enabled where is it not supported.